Week5/MyFavoriteMovies/app/src/main/java/com/raywenderlich/myfavoritemovies/model/RoomRepository.kt

```diff
@@ -0,0 +1,47 @@
+package com.raywenderlich.myfavoritemovies.model
+
+import android.os.AsyncTask
+import androidx.lifecycle.LiveData
+
+class RoomRepository : MovieRepository {
+    private val dao: MovieDao = MovieApplication.database.movieDao()
+    private val allMovies: LiveData<List<Movie>>
+
+    init {
+        allMovies = dao.getAllMovies()
```

Comment:
I'd avoid init blocks in these cases, if you can use `lazy`.

Also, it doesn't make sense to fetch a LiveData in the repository, and then not use the repository at all. I get that you've tried to make this a bit nicer, but you've kind of overcomplicated it a bit, with the extra AsyncTasks, and hardcoding the dependencies you need, using init blocks etc.

Don't create the MovieDao here, pass it in the constructor as a parameter - remember the Dependency Inversion principle.

Also, you don't necessarily need AsyncTasks, you can use `allowMainThreadQueries()` (which you'll work with in the Week 6 HW).